.github/workflows/update-metadata-glsa.yml

```
@@ -0,0 +1,58 @@
+name: Keep GLSA metadata updated
+on:
+  schedule:
+    - cron:  '0 7 1 * 1'
+  workflow_dispatch:
+
+jobs:
+  keep-glsa-metadata-updated:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout portage-stable
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: ./portage-stable
+      - name: Checkout Gentoo
+        uses: actions/checkout@v3
+        with:
+          repository: gentoo-mirror/gentoo
+          path: gentoo
+          fetch-depth: 1
+          ref: master
+      - name: Checkout build scripts
+        uses: actions/checkout@v3
+        with:
+          repository: flatcar/flatcar-build-scripts
+          path: flatcar-build-scripts
+      - name: Update GLSA metadata
+        id: update-glsa-metadata
+        run: |
+          git config --global user.name "Flatcar Buildbot"
+          git config --global user.email "buildbot@flatcar-linux.org"
+          old_head=$(git -C portage-stable rev-parse HEAD)
+          # need to rsync binary data Manifest* and timestamp* which are not in the Git repos.
+          rsync --archive rsync://rsync.gentoo.org/gentoo-portage/metadata/glsa/{Manifest,timestamp}* ./gentoo/metadata/glsa/
+          cd portage-stable
+          GENTOO_REPO=../gentoo ../flatcar-build-scripts/sync-with-gentoo "metadata/glsa"
+          cd ..
+          new_head=$(git -C portage-stable rev-parse HEAD)
+          updated=0
+          if [[ "${new_head}" != "${old_head}" ]]; then
+              updated=1
+          fi
+          todaydate=$(date +%Y-%m-%d)
+          echo "UPDATED=${updated}" >>"${GITHUB_OUTPUT}"
+          echo "TODAYDATE=${todaydate}" >>"${GITHUB_OUTPUT}"
```

I'm wondering if we even need this. The value of sync-with-gentoo is that it will also find the last commit in Gentoo repo that modified the synced package. GLSA stuff comes from outside the repo, so it won't be a part of the git history. I suppose you could do:

```suggestion
          cd portage-stable # probably won't be needed if we only clone portage-stable above
          rm -rf metadata/glsa # not sure if this is needed too, rsync maybe will remove old stuff
          # need to rsync binary data Manifest* and timestamp* which are not in the Git repos.
          rsync --archive rsync://rsync.gentoo.org/gentoo-portage/metadata/glsa/{Manifest,timestamp}* ./metadata/glsa/
          todaydate=$(date +%Y-%m-%d)
          echo "TODAYDATE=${todaydate}" >>"${GITHUB_OUTPUT}"
```

That way, you wouldn't even need to clone the Gentoo repo. You could keep the clone of portage-stable as a current working directory.

---

simplified with some more changes